pommerman/network/ion_client/__init__.py

```diff
@@ -0,0 +1,219 @@
+    ui.info(
+        Strings.server_connected.value, ui.yellow, Strings.server_players.value,
+        str(status[0]) + ",", Strings.server_matches.value, status[1]
+    )
+    intent(network)
+
+
+def intent(network):
+    """Arguments:   
+    * network: An `network`(pommerman.network.ion_client.network) object  
+    Description: This creates a prompt for the user where they can choose to:  
+    * Play a match  
+    * Create/Join a room  
+    * Replay a match  
+    * Exit the application
+    """
+    i = ui.ask_choice(
+        Strings.intent.value, [
+            Strings.intent_match.value, Strings.intent_room.value,
+            Strings.intent_replay.value, Strings.intent_exit.value
+        ]
+    )
+    if i == Strings.intent_match.value:
+        match(network)
+    elif i == Strings.intent_room.value:
+        match(network, True)
+    elif i == Strings.intent_replay.value:
+        replay(network)
+    elif i == Strings.intent_exit.value:
+        exit(0)
+
+
+def match(
+    network, room=False, import_name=False, agent_class=False, ui_en=True
+):
+    """Arguments:  
+    * network: An `network`(pommerman.network.ion_client.network) object  
+    * room: The room to be created/joined. If False, the public room will be \
+joined  
+    * import_name: The module which contains your agent  
+    * agent_class: The class of the agent within the module  
+    * ui_en: If the UI is enabled or disabled (This also controls if exception \
+are raised or not)  
+    Description: This facilitates playing a match"""
+    # Note: Imports are local as they aren't required anywhere else
+    from os import getcwd
+    from sys import path
+    from importlib import import_module
+    from gym.spaces import Discrete
+    from pommerman.constants import Item
+    if room:
+        room = str(ui.ask_string(Strings.room_code.value))
+    # Info: The Current Working Directory has to be added to path to import modules from it
+    path.append(getcwd())
+    if not (import_name and agent_class) and ui_en:
+        agent = import_module(ui.ask_string(Strings.match_import.value))
+        agent_class = ui.ask_string(Strings.match_class_name.value)
+    else:
+        agent = import_module(import_name)
+    if agent_class not in agent.__dir__():
+        if ui_en:
+            ui.fatal(Strings.error_invalid_class.value)
+        else:
+            raise Exception(constants.Exceptions.match_invalid_class.value)
+    agent = getattr(agent, agent_class)
+    if getattr(agent, "act") and ui_en:
+        ui.info(ui.green, Strings.match_agent_success.value)
+    agent = agent()
+    if ui_en:
+        ui.info(ui.yellow, Strings.server_comm.value)
+    try:
+        network.join_list(room)
+    except Exception as e:
+        if ui_en:
+            ui.fatal(e)
+        raise e
+    if ui_en:
+        ui.info(Strings.match_variant.value, ui.yellow, network.mode)
+        ui.info(ui.yellow, Strings.match_wait.value)
+    try:
+        network.waitm()
```

Comment:
Does it matter if it receives the request in order? It could be that there are a bunch of join_list calls and then a waitm call that receives back the 3rd join_list call instead of the 1st. Will that cause problems?